### PR TITLE
[NVIDIA] Remove logic of combining bias and mask

### PR DESCRIPTION
**Pull request by kaixih**
The cudnn API has already supported the combination of bias and mask from [this PR](https://github.com/google/jax/pull/22078). We are removing the logic from the public sdpa API and pass the mask directly.

cc. @Cjkkkk

### CODE REVIEW

**Comment by kaixih** [comment]
@superbobry @sbodenstein for review. Thx.

**Comment by kaixih** [comment]
Gentle ping @superbobry

**Comment by superbobry** [comment]
Please remove the `# type: ignore` comment mypy complains about.

**Review by superbobry — APPROVED** [review]

File: jax/_src/nn/functions.py [hunk 1]
```diff
@@ -907,26 +907,8 @@ def _check_has_shape(t: Array, shape: Sequence[int], name: str) -> None:
...
-      # TODO(kaixih@nvidia): The logic should be moved to the internal of
-      # cudnn_dot_product_attention.
-      if bias is None:
-        bias = mask
-      elif mask is not None:
-        bias = bias + mask
-
       return cudnn_dot_product_attention(
-          query, key, value, bias, mask=None, scale=scale_val,  # type: ignore[arg-type]
+          query, key, value, bias, mask, scale=scale_val,  # type: ignore[arg-type]
```
[the remark this answers is not in the document]
Comment (kaixih) [reply jax/_src/nn/functions.py]: Sorry, I missed this comment. Right, I don't think we need this anymore. Removed. PTAL.

**Comment by superbobry** [comment]
Please squash the commit, so that we can merge.

**Comment by kaixih** [comment]
> Please squash the commit, so that we can merge.

Done. PTAL.

**Pull request closed without merging**